content/docs/1.2.4/volumes-and-nodes/volume-size.md

```diff
@@ -19,29 +19,90 @@ In this section, you'll have a better understanding of concepts related to volum
+
+{{< figure src="/img/screenshots/volumes-and-nodes/volume-size-illustration-fig2.png" >}}
+
+4. Then, rewrite the 4 Gi data (data#1), and the `df` command in the filesystem shows 4 Gi used space again. Howerver, the `actual size` is increased by 4 Gi and becomes 8.25Gi. See Figure 3(a) of the illustration.
+
+     > After deletion, filesystem may or maynot reuse the recently freed blocks from recently deleted files according to the filesystem design and please refer to [Block allocation strategies of various filesystems](https://www.ogris.de/blkalloc). If the volume nominal `size` is 12 Gi, the `actual size` in the end would range from 4 Gi to 8 Gi since the filesystem may or maynot reuse the freed blocks. On the other hand, if the volume nominal `size` is 6 Gi, the `actual size` at the end would range from 4 Gi to 6 Gi, because the filesystem has to reuse the freed blocks in the 2nd round of writing. See Figure 3(b) of the illustration.
+     >
+     > Thus, allocating an appropriate nominal `size` for a volume that holds heavy writing tasks according to the IO pattern would make disk space usage more efficient.
+
+{{< figure src="/img/screenshots/volumes-and-nodes/volume-size-illustration-fig3.png" >}}
+
+5. Take a snapshot (snapshot#1). See Figure 4 of the illustration.
+    - Now data#1 is stored in snapshot#1.
+    - The new volume head size is almost 0.
+    - With the volume head and the snapshot included, the `actual size` remains 8.25 Gi.
+
+{{< figure src="/img/screenshots/volumes-and-nodes/volume-size-illustration-fig4.png" >}}
+
+6. Delete data#1 from the mount point.
     - The data#1 filesystem level removal info is stored in current volume head file. For snapshot#1, data#1 is still retained as the historical data.
-    - The `actual size` is still 2Gi.
-{{< figure src="/img/screenshots/volumes-and-nodes/volume-size-example-2.png" >}}
-4. Write 4Gi data (data#2) in the volume mount, then take one more snapshot (snapshot#2). See Figure 3 of the illustration.
-    - Now the `actual size` is 6Gi, which is greater than the nominal `size`.
-    - There is overlapping between 2 snapshots in block level (See the 2 snapshots in Figure 3), since the data#1 is marked as removed in snapshot#2 then the filesystem reuses the space.
-{{< figure src="/img/screenshots/volumes-and-nodes/volume-size-example-3.png" >}}
-5. Delete snapshot#1 and wait for snapshot purge complete. See Figure 4 of the illustration.
+    - The `actual size` is still 8.25 Gi.
+
+7. Write 8 Gi data (data#2) in the volume mount, then take one more snapshot (snapshot#2). See Figure 5 of the illustration.
+    - Now the `actual size` is 16.3 Gi, which is greater than the volume nominal `size`.
+    - There is overlapping between 2 snapshots in block level (See the 2 snapshots in Figure 6), since the data#1 is marked as removed in snapshot#2 then the filesystem reuses the space.
```

Comment:
From a filesystem's perspective, the overlapping part (between the 2 snapshots) is considered as the blocks that have to be reused or overwritten. But in terms of Longhorn, these blocks are actually fresh ones held in another snapshot/volume head. See the 2 snapshots in Figure 6.